### DISCUSSION TITLE
Is there a way to get a callback when window closes?

### DISCUSSION THREAD

**Discussion by myyrakle** [fltk-rs/fltk-rs#1338] [Q&A] [answered]
I'm using a background thread separate from fltk for separate processing.

So I need to detect and close it when I press the x button in window to exit.

Is there a supported way to solve this?

**Comment by MoAlyousef** [comment, marked as answer]
A window's callback is triggered when you try to close it.
So you can use window.set_callback() and it would trigger when you try to close the window. 
However, you would still need to handle closing when you use the window's set_callback method:
```rust
win.set_callback(|w| {
    // handle custom cleanup
    if app::event() == Event::Close {
        w.hide();
    }
});
```

**Reply by myyrakle** [reply]
Thanks for your kind answer!
